### PR TITLE
OCM-6053 | feat: Day1 SG support for HCP

### PR DESCRIPTION
**Pull request by hunterkepley**
Supports adding SGs to HCP clusters in day 1

Currently, CS has a disabled feature toggle, so this change allows the feature to work, but it will not work until the toggle is enabled

### CODE REVIEW

**Comment by hunterkepley** [comment]
Do not merge until https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/8611 is merged- i believe it will break HCP cluster creation otherwise

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -2571,20 +2571,23 @@ func run(cmd *cobra.Command, _ []string) {
...
-	getSecurityGroups(r, cmd, isVersionCompatibleComputeSgIds,
-		securitygroups.InfraKind, useExistingVPC, isHostedCP, subnets,
-		subnetIDs, &additionalInfraSecurityGroupIds)
-
 	additionalControlPlaneSecurityGroupIds := args.additionalControlPlaneSecurityGroupIds
-	getSecurityGroups(r, cmd, isVersionCompatibleComputeSgIds,
-		securitygroups.ControlPlaneKind, useExistingVPC, isHostedCP, subnets,
-		subnetIDs, &additionalControlPlaneSecurityGroupIds)
+	if !isHostedCP {
+		getSecurityGroups(r, cmd, isVersionCompatibleComputeSgIds,
```
Comment (gdbranco) [inline cmd/create/cluster/cmd.go:2583]: I think you may now need to check if the flag changed for these two sets and return an error to the customer if it is hostedCp

Comment (hunterkepley) [reply cmd/create/cluster/cmd.go:2583]: good catch, will add that along with any other requested changes soon

File: cmd/describe/cluster/cmd.go [hunk 2]
```diff
@@ -743,10 +743,17 @@ Nodes:
...
 	}
+	hasSgsCompute := len(cluster.AWS().AdditionalComputeSecurityGroupIds()) > 0
 	hasSgsControlPlane := len(cluster.AWS().AdditionalControlPlaneSecurityGroupIds()) > 0
 	hasSgsInfra := len(cluster.AWS().AdditionalInfraSecurityGroupIds()) > 0
-	if hasSgsControlPlane || hasSgsInfra {
+	if hasSgsControlPlane || hasSgsInfra || hasSgsCompute {
 		nodeConfig += " - Additional Security Group IDs:\n"
+		if hasSgsCompute {
+			nodeConfig += fmt.Sprintf(
+				"   - Compute:		%s\n",
```
Comment (gdbranco) [inline cmd/describe/cluster/cmd.go:753]: Oh this is intentionally left out because the customer is supposed to check it from describe machine pool :)

**Pull request merged**